internal/pkg/db/scripts/types.sql

```diff
@@ -0,0 +1,374 @@
+INSERT OR IGNORE INTO purpose (id, name) VALUES (15, 'INSTALL');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (16, 'LIBRARY');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (17, 'MACHINE_LEARNING_MODEL');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (18, 'MANIFEST');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (19, 'MODEL');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (20, 'MODULE');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (21, 'OPERATING_SYSTEM');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (22, 'OTHER');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (23, 'PATCH');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (24, 'PLATFORM');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (25, 'REQUIREMENT');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (26, 'SOURCE');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (27, 'SPECIFICATION');
+INSERT OR IGNORE INTO purpose (id, name) VALUES (28, 'TEST');
+
+-- @label Define triggers
+CREATE TRIGGER IF NOT EXISTS tr_purpose_before_update
+BEFORE UPDATE ON purpose
+BEGIN
+    SELECT RAISE(IGNORE);
+END;
+
+CREATE TRIGGER IF NOT EXISTS tr_purpose_before_insert
+BEFORE INSERT ON purpose
+BEGIN
+    SELECT RAISE(IGNORE);
+END;
+
+-- @block create `node_type` table
+-- @conn bomctl
+-- @label Initialize `node_type` table
+-- #----------------------------------------------#
+-- #----[ NodeType enum ]-------------------------#
+-- #----------------------------------------------#
+-- @label Create table
+CREATE TABLE IF NOT EXISTS node_type (
+    id   INTEGER PRIMARY KEY,
+    name TEXT
+);
+
+-- @label Populate values
+INSERT OR IGNORE INTO node_type (id, name) VALUES (0, 'PACKAGE');
+INSERT OR IGNORE INTO node_type (id, name) VALUES (1, 'FILE');
+
+-- @label Define triggers
+CREATE TRIGGER IF NOT EXISTS tr_node_type_before_update
+BEFORE UPDATE ON node_type
+BEGIN
+    SELECT RAISE(IGNORE);
+END;
+
+CREATE TRIGGER IF NOT EXISTS tr_node_type_before_insert
+BEFORE INSERT ON node_type
+BEGIN
+    SELECT RAISE(IGNORE);
+END;
+
+-- @block create `edge_type` table
+-- @conn bomctl
+-- @label Initialize `edge_type` table
+-- #----------------------------------------------#
+-- #----[ EdgeType enum ]-------------------------#
+-- #----------------------------------------------#
+-- @label Create table
+CREATE TABLE IF NOT EXISTS edge_type (
+    id   INTEGER PRIMARY KEY,
+    name TEXT
+);
+
+-- @label Populate values
+INSERT OR IGNORE INTO edge_type (id, name) VALUES (0, 'UNKNOWN');
+INSERT OR IGNORE INTO edge_type (id, name) VALUES (1, 'amends');
+INSERT OR IGNORE INTO edge_type (id, name) VALUES (2, 'ancestor');
+INSERT OR IGNORE INTO edge_type (id, name) VALUES (3, 'buildDependency');
+INSERT OR IGNORE INTO edge_type (id, name) VALUES (4, 'buildTool');
+INSERT OR IGNORE INTO edge_type (id, name) VALUES (5, 'contains');
+INSERT OR IGNORE INTO edge_type (id, name) VALUES (6, 'contained_by');;
+INSERT OR IGNORE INTO edge_type (id, name) VALUES (7, 'copy');
```

Comment:
Strictly speaking, we don't need the enum labels to build the cache. If they get generated automatically it's fine but if we're maintaining the SQL manually it would make sense to drop these.